src/app/api/users.ts

```diff
@@ -24,10 +29,92 @@ export async function getUsers(): Promise<UserResponse> {
     }
 
     const body: UserResponse = await response.json();
-    console.log(body);
     return body;
   } catch (error) {
-    console.error('Failed to get data : ', error);
     throw error;
   }
 }
+
+// 내 정보 조회
+export async function getUserInfo(): Promise<UserInfoResponse | ErrorMessage> {
+  const token = getCookie('accessToken');
+  const obj: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  };
+  try {
+    const res = await fetch(`${API_URL}/users/me`, obj);
+    const data = await res.json();
+
+    if (!res.ok) throw new Error(`${data.message}`);
+
+    return data;
+  } catch (err) {
+    if (err instanceof Error) {
+      return { message: err.message };
+    }
+    return { message: '이미지 업로드 중 알 수 없는 오류가 발생했습니다.' };
+  }
+}
+
+// 이미지 업로드 API
+export async function uploadProfileImage(
+  image: File,
+): Promise<ProfileImageResponse | ErrorMessage> {
+  const token = getCookie('accessToken');
+
+  const formData = new FormData();
+  formData.append('image', image);
+  const obj: RequestInit = {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+    body: formData,
+  };
+
+  try {
+    const res = await fetch(`${API_URL}/users/me/image`, obj);
+    const data = await res.json();
+
+    if (!res.ok) throw new Error(`${data.message}`);
+
+    return data;
+  } catch (err) {
+    if (err instanceof Error) {
+      return { message: err.message };
+    }
+    return { message: '이미지 업로드 중 알 수 없는 오류가 발생했습니다.' };
+  }
+}
+
+// 내 정보 수정
+export async function editUserInfo(
+  userData: UpdateUser,
+): Promise<ErrorMessage> {
```

Comment:
지금은 수정 후 리스폰스를 사용하지 않아서 타입 에러가 나지 않지만 나중에 사용하신다면 타입 에러가 날 것 같아요!

`@planit-types`에 아래와 같이 타입을 정의하고 반환 타입에 추가해주시면 리스폰스에 대한 타입 에러가 나지 않을 것 같습니다~

```ts
export type EditUserResponse = {
  id: number;
  email: string;
  nickname: string;
  profileImageUrl: string;
  createdAt: string;
  updatedAt: string;
}
```

```suggestion
): Promise<EditUserResponse | ErrorMessage> {
```

Reply:
해당 타입을 사용하지 않는데 타입을 넣을 필요가 있을까요?

Reply:
함수를 사용하는 쪽에서 반환하는 값이 에러 밖에 없나? 라고 생각이 들 수 있겠지만,
리스폰스 결과값을 사용할 일이 없다면 굳이 타입을 작성하는 수고를 하지 않는 장점이 있겠네요.

요거는 취향 차이인 것 같아요! 혜선님께서 판단해서 해주세용

Reply:
근데 권희님 말씀이 맞는거 같아요 
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ
바로 수정 긔